### PR TITLE
Wire in TCK, finish implementing header kind, and fix some bugs

### PR DESCRIPTION
**Pull request by anuraaga**

### CODE REVIEW

**Review by codefromthecrypt — APPROVED** [review]
works for me. deferring in case @jcchavezs has any feedback as I haven't done any JS in a couple years.

File: packages/express/test/tck.ts [hunk 1]
```diff
@@ -0,0 +1,107 @@
...
+import express from 'express';
+import { GenericContainer, TestContainers } from 'testcontainers';
+import { assert } from 'chai';
+
+import httpwasm from '../index.js';
+
+const tckImage = 'ghcr.io/http-wasm/http-wasm-tck:sha-e25f1a2';
+
+describe('tck', async function () {
+  this.timeout(10000);
```
Comment (jcchavezs) [inline packages/express/test/tck.ts]: Probably not too important but why this number? why not 100? why not 100000? Also, is this secs? Maybe better to use a constant.

File: packages/express/test/tck.ts [hunk 2]
```diff
@@ -0,0 +1,107 @@
...
+      for (const [key, value] of Object.entries(req.headers)) {
+        if (!value) {
+          continue;
+        }
+        const values = Array.isArray(value) ? value : [value];
+        for (let i = 0; i < values.length; i++) {
+          res.setHeader(`x-httpwasm-next-header-${key}-${i}`, values[i]);
+        }
+      }
+      res.sendStatus(200);
```
Comment (jcchavezs) [inline packages/express/test/tck.ts:53]: in general for tests I prefer to use a code different than 200. Some frameworks (maybe express too) can chose to return 200 even if status code hasn't being sent so better to use 202 o 201 to do further assertions.

File: packages/express/test/tck.ts (hunk 1 above)
Comment (anuraaga) [reply packages/express/test/tck.ts]: Extracted a comment - the value is arbitrary, 10s seems to generally be a good number for slower tests based on experience (in Java though)

File: packages/express/test/tck.ts (hunk 2 above)
Comment (anuraaga) [reply packages/express/test/tck.ts:53]: This would need changing the TCK so will leave it for now and think on whether it's useful

**Pull request merged**